pyshelf/resource_identity_factory.py

```diff
@@ -0,0 +1,37 @@
+from pyshelf.resource_identity import ResourceIdentity
+
+
+class ResourceIdentityFactory(object):
+    def __init__(self, artifact_path_builder):
+        self.artifact_path_builder = artifact_path_builder
+
+    def from_resource_url(self, resource_url):
+        """
+            Creates a ResourceIdentity from the full
+            resource url.  For example:
+            /<bucket-name>/artifacts/<path>
+
+            Args:
+                resource_url(basestring)
+
+            Returns:
+                pyshelf.resource_identity.ResourceIdentity
+        """
+        identity = ResourceIdentity(resource_url)
+        return identity
+
+    def from_cloud_identifier(self, cloud_identifier):
```

Comment:
I was going to wait until its used and then let it get tested functionally like we do with other factories and the like